spread/runner.go

```diff
@@ -718,6 +724,38 @@ func (r *Runner) worker(backend *Backend, system *System, order []int) {
 	}
 }
 
+func (r *Runner) fetchProjectArtifacts(client *Client) error {
```

Comment:
most of the code in this function is duplicated from fetchArtifacts. You could create a new function with the code in common and called it from both.